app/src/main/java/com/duckduckgo/app/browser/indonesiamessage/IndonesiaNewTabSectionView.kt

```diff
@@ -0,0 +1,129 @@
+import com.duckduckgo.common.ui.view.gone
+import com.duckduckgo.common.ui.view.show
+import com.duckduckgo.common.ui.viewbinding.viewBinding
+import com.duckduckgo.common.utils.ViewViewModelFactory
+import com.duckduckgo.di.scopes.AppScope
+import com.duckduckgo.di.scopes.ViewScope
+import com.duckduckgo.newtabpage.api.NewTabPageSection
+import com.duckduckgo.newtabpage.api.NewTabPageSectionPlugin
+import dagger.android.support.AndroidSupportInjection
+import javax.inject.Inject
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+
+@InjectWith(ViewScope::class)
+class IndonesiaNewTabSectionView @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyle: Int = 0,
+) : FrameLayout(context, attrs, defStyle) {
+
+    @Inject
+    lateinit var viewModelFactory: ViewViewModelFactory
+
+    private var coroutineScope: CoroutineScope? = null
+
+    private val binding: ViewIndonesiaNewTabSectionBinding by viewBinding()
+
+    private val viewModel: IndonesiaNewTabSectionViewModel by lazy {
+        ViewModelProvider(findViewTreeViewModelStoreOwner()!!, viewModelFactory)[IndonesiaNewTabSectionViewModel::class.java]
+    }
+
+    override fun onAttachedToWindow() {
+        AndroidSupportInjection.inject(this)
+        super.onAttachedToWindow()
+
+        findViewTreeLifecycleOwner()?.lifecycle?.addObserver(viewModel)
+
+        @SuppressLint("NoHardcodedCoroutineDispatcher")
+        coroutineScope = CoroutineScope(SupervisorJob() + Dispatchers.Main)
+
+        viewModel.viewState
+            .onEach { render(it) }
+            .launchIn(coroutineScope!!)
+    }
+
+    private fun render(viewState: ViewState) {
+        if (viewState.showMessage) {
+            showMessage()
+        } else {
+            binding.newTabIndonesiaMessage.gone()
+        }
+    }
+
+    private fun showMessage() {
+        with(binding.newTabIndonesiaMessage) {
+            setMessage(
+                Message(
+                    topIllustration = com.duckduckgo.mobile.android.R.drawable.ic_announce,
+                    title = context.getString(R.string.newTabPageIndonesiaMessageHeading),
+                    subtitle = context.getString(R.string.newTabPageIndonesiaMessageBody),
+                    action = context.getString(R.string.newTabPageIndonesiaMessageCta),
+                ),
+            )
+            onPrimaryActionClicked {
+                gone()
+                viewModel.onMessageDismissed()
+            }
+            onCloseButtonClicked {
+                gone()
+                viewModel.onMessageDismissed()
+            }
+            show()
+            viewModel.onMessageShown()
+        }
+    }
+}
+
+@ContributesActivePlugin(
+    AppScope::class,
+    boundType = NewTabPageSectionPlugin::class,
+    priority = -1,
```

Comment:
this should be moved to `NewTabPageSectionPlugin.PRIORITY_INDONESIA_MESSAGE` with a value of 5.

Reply:
Updated :+1: